File: app/src/main/java/com/example/loding/Login/MainActivity.java
package com.example.loding.Login;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.example.loding.Home.Home;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // 检查登录状态
        SharedPreferences preferences = getSharedPreferences("user_prefs", MODE_PRIVATE);
        boolean isLoggedIn = preferences.getBoolean("is_logged_in", false);

        if (isLoggedIn) {
            // 已登录，跳转到主页
            Intent intent = new Intent(this, Home.class);
            startActivity(intent);
        } else {
            // 未登录，跳转到登录页
            Intent intent = new Intent(this, LoginActivity.class);
            startActivity(intent);
        }

        finish();
    }
}

File: app/src/main/java/com/example/loding/Login/MainActivity.kt
package com.example.loding

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.loding.Home.Home
import com.example.loding.Login.LoginActivity


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // 检查登录状态
        val preferences: SharedPreferences = getSharedPreferences("user_prefs", MODE_PRIVATE)
        val isLoggedIn: Boolean = preferences.getBoolean("is_logged_in", false)

        if (isLoggedIn) {
            // 已登录，跳转到主页
            val intent = Intent(this, Home::class.java)
            startActivity(intent)
        } else {
            // 未登录，跳转到登录页
            val intent = Intent(this, LoginActivity::class.java)
            startActivity(intent)
        }
        finish()
    }
}